### ISSUE TITLE
VS Code extension resolves the markuplint module from the workspace root only, ignoring `workingDirectories`

### ISSUE THREAD

**Issue by taku10101** [markuplint/markuplint#4048]
## Summary

`markuplint.workingDirectories` (#1287 / #3288) made the extension find the right **config file** in a monorepo — thank you. However, the **markuplint module itself** is still resolved only from `process.cwd()` (the workspace folder). In a monorepo where markuplint is installed per package rather than at the root, the extension falls back to the bundled copy, and the bundled copy cannot load a parser that lives in the user's workspace, so `.tsx` files get no diagnostics.

I'd like module resolution to honor `workingDirectories` as well.

## Current behavior

`vscode/src/server/server.ts` receives `workingDirectories` (L66) and passes it to the document handlers (L90), but `getModule(log)` (L77) only gets the logger, and it runs once inside `onInitialized`.

`vscode/src/server/get-module.ts`:

```ts
async function fileResolve(log: (message: string) => void) {
  try {
    const modPath = await Files.resolve('markuplint', process.cwd(), process.cwd(), message => log(message));
```

So module resolution ignores both `workingDirectories` and the document's own path, and one module instance is used for every working directory.

## Root cause of the fallback failure

When the local module isn't found, the server falls back to the bundled markuplint. The modules a config refers to (`parser` / `specs` / `plugins` / `extends`) are resolved **from the markuplint installation's location**, not from the config file's directory:

- `resolveNameOrAbsPath()` keeps the bare specifier only when `moduleExists()` returns true, otherwise it rewrites it to `path.resolve(configDir, name)`
- `moduleExists()` uses `import(name)` / `createRequire(import.meta.url)` inside `@markuplint/file-resolver`

The extension bundles only the `astro` / `html` / `svelte` / `vue` parsers, so any workspace-installed parser becomes unreachable once the fallback kicks in, and the specifier is rewritten to a path that does not exist:

```console
$ node <extension>/node_modules/markuplint/bin/markuplint.mjs \
    --config applications/app/.markuplintrc.json applications/app/src/Foo.tsx
Error: Parser module "<abs>/applications/app/@markuplint/jsx-parser" is not found.
```

The same config and the same file lint successfully with the markuplint installed in `applications/app`.

## Reproduction

pnpm workspace, no markuplint at the repository root:

```
repo/
├── applications/app/          .markuplintrc.json + markuplint, @markuplint/jsx-parser, @markuplint/react-spec
├── applications/mobile-app/   (same)
└── packages/foo/              (same)
```

`.vscode/settings.json`:

```json
{
  "markuplint.targetLanguages": ["typescriptreact"],
  "markuplint.workingDirectories": [{ "pattern": "./applications/*/" }, { "pattern": "./packages/*/" }]
}
```

`applications/app/.markuplintrc.json`:

```json
{
  "parser": { "\\.tsx$": "@markuplint/jsx-parser" },
  "specs": { "\\.tsx$": "@markuplint/react-spec" },
  "extends": ["markuplint:recommended-react"]
}
```

Open `repo/` as the workspace folder: the config is resolved correctly, but no diagnostics appear on `.tsx`. The CLI in each package works fine.

## Workaround

Install `markuplint` at the workspace root as well. The extension then loads the local module, and the parser is reachable from it (in pnpm, through the workspace's virtual store).

It works, but it puts a dependency in a package that never runs the tool, which conflicts with declaring dependencies where they are used. The ESLint and Stylelint extensions do not need this.

## Proposal

- Resolve `markuplint` per working directory — use the working directory the document maps to as the resolution base, fall back to the document's own directory when `workingDirectories` is unset, and cache the resolved module per directory.
- At minimum, surface it when the bundled fallback cannot load the parser a config requires. As far as I could tell the file is simply left without diagnostics, so the cause is not obvious from the editor.

## Environment

- markuplint: 5.0.0
- VS Code extension: `markuplint.markuplint-vscode` 5.0.0
- Parser lang: JSX (`.tsx`)
- Package manager: pnpm (workspace)
- OS: macOS

## Related

- #1287 / #3288 — `workingDirectories` (config resolution)
- #2837 / #2838 — bundled fallback behavior
